src/application.ts

```diff
@@ -10,18 +10,23 @@ import {
 import {
   Simple,
   templateFactory,
-  RenderResult
+  RenderResult,
+  ComponentDefinition,
+  Component
 } from '@glimmer/runtime';
 import {
   UpdatableReference
 } from '@glimmer/object-reference';
+import {
+  Option
+} from '@glimmer/util';
 import ApplicationRegistry from './application-registry';
 import DynamicScope from './dynamic-scope';
 import Environment from './environment';
+import mainTemplate from './templates/main';
```

Comment:
TS complains about this, since this module doesn't exist until we start compiling. I think an approach like with module map is in order, but didn't want to block this whole PR on this point.